cmd/cluster/server.go

```diff
@@ -21,44 +24,56 @@ type Cluster struct {
 
 // newCluster will create a new Cluster with default values.
 func newCluster() *Cluster {
+	// 1.1.1.1 is used for route lookup, to determine the local interface that would
+	// be used to reach this address.  It doesn't actually talk to that address.
 	local, err := nodes.LocalAddress("1.1.1.1:1")
-
 	if err != nil {
 		return nil
 	}
 
 	return &Cluster{
 		RedisAddr:      "127.0.0.1:6379",
 		Namespace:      "namespace",
-		Target:         local.String() + ":8125",
+		Target:         local.String(),
 		UpdateInterval: time.Second,
-		ExpiryInterval: 30 * time.Second,
+		ExpiryInterval: 4 * time.Second,
 	}
 }
 
 // AddFlags adds flags for a specific Server to the specified FlagSet.
 func (c *Cluster) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&c.RedisAddr, "redis-addr", c.RedisAddr, "Redis address")
 	fs.StringVar(&c.Namespace, "namespace", c.Namespace, "Namespace")
-	fs.StringVar(&c.Target, "target", c.Target, "Target port")
+	fs.StringVar(&c.Target, "target", c.Target, "Target host to advertise")
 	fs.DurationVar(&c.UpdateInterval, "update-interval", c.UpdateInterval, "Cluster update interval")
 	fs.DurationVar(&c.ExpiryInterval, "expiry-interval", c.ExpiryInterval, "Cluster expiry interval")
 }
 
 // Run runs the specified Cluster.
-func (c *Cluster) Run() error {
-	rnt := nodes.NewRedisNodeTracker(c.RedisAddr, c.Namespace, c.Target, c.UpdateInterval, c.ExpiryInterval)
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	go rnt.Run(ctx)
+func (c *Cluster) Run(ctx context.Context) {
+	picker := nodes.NewConsistentNodePicker(c.Target, 20)
+
+	options := &redis.Options{
+		Addr: c.RedisAddr,
+		DB:   0,
+	}
+
+	redisClient := redis.NewClient(options)
```

Comment:
I can't see it here, but it is worth using a universal client here from `go-redis` so that you can use single host or clustered redis?

Reply:
/nods

I'll look in to it.  For now I just want to get this merged.